index.js

```diff
@@ -211,23 +213,37 @@ module.exports = () => {
 		};
 	};
 
+	const drainOpenSubscriptions = async () => {
+		let i = openSubscriptions.length;
+		while (i--) {
+			debug('Closing active subscription');
+			await openSubscriptions[i].close();
+			openSubscriptions.splice(i, 1);
+		}
+	};
+
 	const stop = async () => {
+		debug('Closing open stream subscriptions and draining...');
+		await drainOpenSubscriptions();
+
+		let timer;
+		// eslint-disable-next-line no-return-assign
+		const checkIfSubscriptionIsEmpty = () => new Promise(resolve => timer = setInterval(async () => {
```

Comment:
Maybe I'm a bit rusty but why rewriting this function? I think the previous implementation is even more readable...

Reply:
The point is the function was not allowing the component to shutdown cleanly. The timer was continuing as it wasn’t cleaned up. The rest of the changes reflect that the function needs to let all messages complete **before** cleaning and closing connections.